From ffc11046c996e17647cece5b89d8e7bd9e3e64ee Mon Sep 17 00:00:00 2001
From: Valentine Barshak <vbarshak@dev.rtsoft.ru>
Date: Thu, 2 May 2013 12:13:37 +0400
Subject: [PATCH 076/127] arm: tegra: ktt30: Make bootloader framebuffer
 checks more strict

Add bootloader framebuffer address and size checks. Aslo add
address checks for framebuffer2.

Signed-off-by: Valentine Barshak <vbarshak@dev.rtsoft.ru>
---
 arch/arm/mach-tegra/board-ktt30-panel.c |    8 ++++++--
 1 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-tegra/board-ktt30-panel.c b/arch/arm/mach-tegra/board-ktt30-panel.c
index 3b0bad7..f615592 100644
--- a/arch/arm/mach-tegra/board-ktt30-panel.c
+++ b/arch/arm/mach-tegra/board-ktt30-panel.c
@@ -758,8 +758,12 @@ int __init ktt30_panel_init(void)
 	res->end = tegra_fb_start + tegra_fb_size - 1;
 
 	/* Copy the bootloader fb to the fb. */
-	tegra_move_framebuffer(tegra_fb_start, tegra_bootloader_fb_start,
+	if (tegra_bootloader_fb_start && tegra_bootloader_fb_size)
+		tegra_move_framebuffer(tegra_fb_start,
+				tegra_bootloader_fb_start,
 				min(tegra_fb_size, tegra_bootloader_fb_size));
+	else
+		tegra_clear_framebuffer(tegra_fb_start, tegra_fb_size);
 
 	if (!err)
 		err = nvhost_device_register(&ktt30_disp1_device);
@@ -776,7 +780,7 @@ int __init ktt30_panel_init(void)
 	 * If the bootloader fb2 is valid, copy it to the fb2, or else
 	 * clear fb2 to avoid garbage on dispaly2.
 	 */
-	if (tegra_bootloader_fb2_size)
+	if (tegra_bootloader_fb2_start && tegra_bootloader_fb2_size)
 		tegra_move_framebuffer(tegra_fb2_start,
 			tegra_bootloader_fb2_start,
 			min(tegra_fb2_size, tegra_bootloader_fb2_size));
-- 
1.7.4.4

